src/servers/src/grpc/otlp.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::result::Result as StdResult;
+
+use opentelemetry_proto::tonic::collector::metrics::v1::metrics_service_server::MetricsService;
+use opentelemetry_proto::tonic::collector::metrics::v1::{
+    ExportMetricsServiceRequest, ExportMetricsServiceResponse,
+};
+use opentelemetry_proto::tonic::collector::trace::v1::trace_service_server::TraceService;
+use opentelemetry_proto::tonic::collector::trace::v1::{
+    ExportTraceServiceRequest, ExportTraceServiceResponse,
+};
+use session::context::{QueryContextBuilder, QueryContextRef};
+use tonic::{Request, Response, Status};
+
+use crate::query_handler::OpenTelemetryProtocolHandlerRef;
+
+pub struct OtlpService {
+    handler: OpenTelemetryProtocolHandlerRef,
+}
+
+impl OtlpService {
+    pub fn new(handler: OpenTelemetryProtocolHandlerRef) -> Self {
+        Self { handler }
+    }
+}
+
+#[async_trait::async_trait]
+impl TraceService for OtlpService {
+    async fn export(
+        &self,
+        request: Request<ExportTraceServiceRequest>,
+    ) -> StdResult<Response<ExportTraceServiceResponse>, Status> {
+        let (_headers, extensions, req) = request.into_parts();
+
+        let ctx = extensions
+            .get::<QueryContextRef>()
+            .cloned()
+            .unwrap_or_else(|| QueryContextBuilder::default().build());
```

Comment:
```suggestion
        let ctx = extensions
            .get::<QueryContextRef>()
            .cloned().context(QueryContextMissingSnafu)?;
```

How about using context here, since we SHOULD insert QueryContext in Extension with or without UserProvider?